### ISSUE TITLE
CSS declaration order

### ISSUE THREAD

**Issue by HouCoder** [juwai/style-guide#32]
I discovered few CSS declaration order rules:
1. [Code Guide by @mdo #CSS declaration order](http://mdo.github.io/code-guide/#css-declaration-order).
2. [idiomatic-css#declaration-order](https://github.com/necolas/idiomatic-css#declaration-order).
3. [Google HTML/CSS Style Guide#Declaration Order](https://google.github.io/styleguide/htmlcssguide.xml?showone=Declaration_Order#Declaration_Order)
4. [Airbnb SCSS Ordering of property declarations](https://github.com/airbnb/css#ordering-of-property-declarations)

Basically just 2 way:
1. Order by group(Box-model, Visual, Positioning). In this way, more transparency in code and easier to understand the layout/behavior just by reading the code, the negative part of this way is hard to start, the development progress maybe slower at the beginning.
2. Order by alphabet.Order by alphabet is pretty common in everywhere,it's brain-dead easy, the negative part is sometimes it make no sense in CSS, like:

``` css
.foo {
  border: 1px solid #000;
  height: 30px;
  line-height: 14px;
  margin: 0 auto;
  overflow: hidden;
  padding: 1px;
  width: 50px;
  max-height: 50px;
  max-width: 100px;
}
```

For readability reason, the `with` & `height` settings should be ordered like this:

``` css
.foo {
  /** ... **/
  height: 30px;
  max-height: 50px;
  width: 50px;
  max-width: 100px;
  /** ... **/
}
```

Personally, I prefer order by group by follow the [Code Guide by @mdo #CSS declaration order](http://mdo.github.io/code-guide/#css-declaration-order).

@juwai/site-dev What you guys think?

**Comment by arkhi** [comment]
This is the related issue on Jira: http://jira.juwai.com/browse/JUW-1662

I’m glad we agree on the grouping. :)

I made a proposal last month: https://juwai.slack.com/files/arkhi/F0Q1Y91TJ/juwai-csscomb.json

The thing that looks weird to me when looking at mdo grouping is that he doesn’t consider `border` to be part of [the box model](https://css-tricks.com/the-css-box-model/). That’s really odd.

Since we’re already using Idiomatic, I would [go for it](https://github.com/necolas/idiomatic-css#declaration-order) if we have to pick one, with the added missing properties.

I like the document for Airbnb as it takes preprocessors into account. That being said, we started adding `@include` and `@extend` at the start. I think it’s good to keep potential bugs at the top. ;)
